### PR TITLE
Frontend rules

### PR DESCRIPTION
**Pull request by sadlil**
ref: #441

Frontend rules specifies a set of rules that should be applied in
HAProxy frontend configuration. The set of keywords are from here
https://cbonte.github.io/haproxy-dconv/1.7/configuration.html#4.1
Only frontend sections can be applied here. It is up to user to
provide valid set of rules.
This allows acls or other options in frontend sections in HAProxy config.
Frontend rules will be mapped with Ingress Rules according to port.

### CODE REVIEW

File: api/ingress.go [hunk 1]
```diff
@@ -160,6 +168,8 @@ type TCPIngressRuleValue struct {
 	// Specifies the node port of the referenced service.
 	NodePort intstr.IntOrString `json:"nodePort,omitempty"`
 
+	FrontendRule []string `json:"frontendRule,omitempty"`
```
Comment (tamalsaha) [inline api/ingress.go]: Remove this?

File: hack/docker/voyager/templates/tcp-frontend.cfg [hunk 2]
```diff
@@ -1,4 +1,9 @@
 frontend {{ .FrontendName }}
 	bind *:{{ .Port }} {{ if .AcceptProxy }}accept-proxy{{ end }} {{ if .SecretName }}ssl no-sslv3 no-tlsv10 no-tls-tickets crt /etc/ssl/private/haproxy/{{ .SecretName }}.pem{{ end }} {{ if .ALPNOptions }}{{ .ALPNOptions }}{{ end }}
 	mode tcp
+
+	{{- range $rule := .FrontendRules }}
```
Comment (tamalsaha) [inline hack/docker/voyager/templates/tcp-frontend.cfg]: Fix indentation

File: hack/docker/voyager/templates/http-frontend.cfg [hunk 3]
```diff
@@ -12,6 +12,11 @@ frontend {{ .FrontendName }}
 	option httplog
 	option forwardfor
 
+    {{- range $rule := .FrontendRules }}
+    {{ $rule -}}
+    {{ end }}
+
```
Comment (tamalsaha) [inline hack/docker/voyager/templates/http-frontend.cfg]: `{{range pipeline}} T1 {{else}} T0 {{end}}`

Should we not inject any custom rule when Frontend Rules are used. Then user has full control.
```
    {{- range $rule := .FrontendRules }}
    {{ $rule }}
    {{ else }}
 	option httplog
  	option forwardfor
    {{ end }}
```

File: hack/docker/voyager/templates/tcp-frontend.cfg (hunk 2 above)
Comment (sadlil) [reply hack/docker/voyager/templates/tcp-frontend.cfg]: blank line will be removed from the generated config.

**Pull request merged**